variables.tf

```diff
@@ -1,9 +1,176 @@
+  type        = string
+  default     = null
+}
+variable "reloader_resources_to_ignore" {
+  description = "List of comma separated resources to ignore for reloader. If multiple are provided they are combined with the AND operator"
+  type        = string
+  default     = null
+}
+variable "reloader_namespaces_selector" {
+  description = "List of comma separated label selectors, if multiple are provided they are combined with the AND operator"
+  type        = string
+  default     = null
+}
+variable "reloader_resource_label_selector" {
+  description = "List of comma separated label selectors, if multiple are provided they are combined with the AND operator"
+  type        = string
+  default     = null
+}
+variable "reloader_ignore_secrets" {
+  description = "Whether to ignore secret changes or not"
+  type        = bool
+  default     = false
+}
+variable "reloader_ignore_configmaps" {
+  description = "Whether to ignore configmap changes or not"
+  type        = bool
+  default     = false
+}
+variable "reloader_is_openshift" {
+  description = "Enable OpenShift DeploymentConfigs"
+  type        = bool
+  default     = true
+}
+variable "reloader_is_argo_rollouts" {
+  description = "Enable Argo Rollouts"
+  type        = bool
+  default     = false
+}
+variable "reloader_reload_on_create" {
+  description = "Enable reload on create events"
+  type        = bool
+  default     = true
+
+}
+variable "reloader_sync_after_restart" {
+  description = "Enable sync after Reloader restarts for Add events, works only when reloadOnCreate is true"
+  type        = bool
+  default     = true
+
+}
+variable "reloader_pod_monitor_metrics" {
+  description = "Enable to scrape Reloader's Prometheus metrics"
+  type        = bool
+  default     = false
+}
+
+variable "reloader_log_format" {
+  description = "The log format to use for reloader. Possible values are `json` or `text`. Default value is `json`"
+  type        = string
+  default     = "text"
+  validation {
+    condition     = contains(["json", "text"], var.reloader_log_format)
+    error_message = "The specified reloader_log_format is not a valid selection! Valid values are `json` or `text`"
+  }
+}
+variable "reloader_custom_values" {
+  description = "String containing custom values to be used for reloader helm chart. See https://github.com/stakater/Reloader/blob/master/deployments/kubernetes/chart/reloader/values.yaml"
+  type        = string
+  default     = null
+}
+
+variable "eso_image_repo" {
+  type        = string
+  description = "The container image repository for External Secrets Operator (ESO). By default, the image is pulled from `ghcr.io/external-secrets/external-secrets`."
+  default     = "ghcr.io/external-secrets/external-secrets"
+}
+
+variable "eso_image_tag_digest" {
+  type        = string
+  description = "The tag digest of the ESO image to deploy. If not provided, a default value will be used."
+  default     = null
+}
```

Comment:
please add variable validation for this. Same as https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/blob/7cfcec01a90be289d6355eae68b8d7bb7df8b4cb/solutions/standard/variables.tf#L376-L379